src/data_access_layer/mappers/event_system/event_queue_mapper.ts

```diff
@@ -0,0 +1,68 @@
+import Result from "../../../common_classes/result"
+import Mapper from "../mapper";
+import {QueryConfig} from "pg";
+import Task from "../../../event_system/task";
+import {plainToClass} from "class-transformer";
+
+const resultClass = Task
+
+/*
+    EventQueueMapper extends the Postgres database Mapper class and allows
+    the user to map a data structure to and from the attached database. The mappers
+    are designed to be as simple as possible and should not contain things like
+    validation or transformation of the data prior to storage - those operations
+    should live in a Repository or on the data structure's class itself. Also
+    try to avoid listing functions, as those are generally covered by the Repository
+    class/interface as well.
+*/
+export default class EventQueueMapper extends Mapper{
+    public static tableName = "queue_tasks";
+
+    private static instance: EventQueueMapper;
+
+    public static get Instance(): EventQueueMapper {
+        if(!EventQueueMapper.instance) {
+          EventQueueMapper.instance = new EventQueueMapper()
+        }
+
+        return EventQueueMapper.instance
+    }
+
+    public async Retrieve(id: string): Promise<Result<Task>> {
+        // TODO verify it's setting the right values on the included event task
+        return super.retrieve<Task>(this.retrieveStatement(id), {resultClass})
+    }
+
+    public async List(): Promise<Task[]> {
```

Comment:
Should List() and listStatement() functionality be moved to a repository?

Reply:
I think this is one of the few cases where we're not going to do that. We don't work with Task's outside of the mapper so doing a repository just for the listing seems a little too much.